tests/unit/pyrflx_test.py

```diff
@@ -1140,30 +1140,116 @@ def test_checksum_value_range(no_conditionals_message: Message) -> None:
+    )
+    tlv_package.set_checksum_functions({"Message": {"Checksum": tlv_checksum_function}})
     data = b"\x09\xff\x01\x00\x02\x01\x02\x00\x00\x00\x00"
     message = refinement_package["Message"]
-    message.set_checksum_function({"Checksum": msg_checksum})
+    message.set_checksum_function({"Checksum": refinement_checksum_function})
     message.parse(data)
     assert message.valid_message
     tlv_message = message.get("Payload")
     assert isinstance(tlv_message, MessageValue)
     assert tlv_message.valid_message
 
 
+def test_set_checksum_to_pyrflx(
+    pyrflx_checksum: PyRFLX,
+    tlv_checksum_function: ty.Callable,
+    refinement_checksum_function: ty.Callable,
+) -> None:
+    # pylint: disable=protected-access
+    pyrflx_checksum.set_checksum_functions(
+        {
+            "Refinement_With_Checksum::Message": {"Checksum": refinement_checksum_function},
+            "TLV_With_Checksum::Message": {"Checksum": tlv_checksum_function},
+        }
+    )
+    refinement_with_checksum_msg = pyrflx_checksum["Refinement_With_Checksum"]["Message"]
+    tlv_with_checksum_msg = pyrflx_checksum["TLV_With_Checksum"]["Message"]
+    assert callable(refinement_with_checksum_msg._checksums["Checksum"].function)
+    assert callable(tlv_with_checksum_msg._checksums["Checksum"].function)
+    assert (
+        refinement_with_checksum_msg._checksums["Checksum"].function.__name__
+        == refinement_checksum_function.__name__
+    )
+    assert (
+        tlv_with_checksum_msg._checksums["Checksum"].function.__name__
+        == tlv_checksum_function.__name__
+    )
+
+
+def test_set_checksum_to_pyrflx_invalid_id(
+    pyrflx_checksum: PyRFLX,
+    tlv_checksum_function: ty.Callable,
+    refinement_checksum_function: ty.Callable,
+) -> None:
+    with pytest.raises(
+        PyRFLXError,
+        match="^pyrflx: error: Refinement_With_Checksum:Message is not a valid identifier:"
+        ' id: error: ":" in identifier parts of "Refinement_With_Checksum:Message"$',
+    ):
+        pyrflx_checksum.set_checksum_functions(
+            {
+                "Refinement_With_Checksum:Message": {"Checksum": refinement_checksum_function},
+                "TLV_With_Checksum::Message": {"Checksum": tlv_checksum_function},
+            }
+        )
+
+    with pytest.raises(
+        PyRFLXError,
+        match="^pyrflx: error: Not_A_Package is not a valid identifier$",
+    ):
+        pyrflx_checksum.set_checksum_functions(
+            {
+                "Not_A_Package": {"Checksum": refinement_checksum_function},
+                "TLV_With_Checksum::Message": {"Checksum": tlv_checksum_function},
+            }
+        )
+
+
+def test_set_checksum_to_pyrflx_package_not_found(
+    pyrflx_checksum: PyRFLX,
+    refinement_checksum_function: ty.Callable,
+) -> None:
+    pyrflx_checksum.set_checksum_functions(
+        {
+            "Not_A_Package::Not_A_Message": {"Checksum": refinement_checksum_function},
+        }
+    )
+
+
```

Comment:
I don't think it's a good idea that there is no error, if a checksum is set for an undefined package.

Reply:
Errors are now raised if a checksum is set for an undefined packge or an undefined message. dce0b75